remapping_commands.sh

```diff
@@ -0,0 +1,195 @@
+
+#generate the flanking sequence intervals
+bedtools slop -i variants.bed -g "$oldgenome".chrom.sizes -b 50 > flanking.bed
+
+#get the fasta sequences for these intervals
+bedtools getfasta -fi ../"$oldgenome" -bed flanking.bed -fo variants_reads.fa
+
+#filter out reads that aren't 101 bases long (origin: variant too close to the start or end of the chromosome)
+awk '($0 ~ ">"){name=$0};($0 !~ ">" && length($0) == 101){print name; print $0}' variants_reads.fa > \
+	variants_read_filtered.fa
+
+#replace the colon separators with "|":
+#storing this information for later on in the script when we split the name by "|" to extract the relevant information 
+#(ALT allele, QUAL, FILT, INFO)
+#this is done because the INFO column can contain ":", which means we wouldn't be able to split by ":", so "|" was 
+#chosen
+sed -i 's/:/|/' variants_read_filtered.fa
+
+#store ref bases
+awk '{print $6}' flanking.bed > old_ref_bases.txt
+
+#store rsIDs
+awk '{print $4}' flanking.bed > rsIDs.txt
+
+#store variant bases
+awk '{print $7}' flanking.bed > variant_bases.txt
+
+#store the other vcf columns
+awk '{print $5, $8, $9}' flanking.bed > qual_filt_info.txt
+
+#paste the names, variant bases, then fasta sequences into a new file
+paste <(grep '^>' variants_read_filtered.fa) old_ref_bases.txt variant_bases.txt rsIDs.txt qual_filt_info.txt \
+  <(grep -v '^>' variants_read_filtered.fa) > temp.txt
+
+#reformat the fasta ID: inconsistencies in the separators, and no new line before the sequence mean that this next 
+#command is a bit ugly
+#Input:
+#>[chr]|[pos interval]   [REF]       [ALT]       [rsID]    [QUAL] [FILT] [INFO]      [seq]
+#replace all spaces and tabs with "|":
+#>[chr]|[pos interval]|[REF]|[ALT|[rsID]|[QUAL|[FILT]|[INFO]|[seq]
+#replace the last "|" with a space (this is between the header and the sequence):
+#>[chr]|[pos interval]|[REF]|[ALT|[rsID]|[QUAL|[FILT]|[INFO] [seq]
+#and finally replace the space with a newline:
+#Output:
+#>[chr]|[pos interval]|[REF]|[ALT|[rsID]|[QUAL|[FILT]|[INFO]
+#[seq]
+sed 's/\t/|/g; s/ /|/g; s/\(.*\)|/\1 /' temp.txt | tr ' ' '\n' > variant_reads.out.fa
+
+
+echo "---------------------------2) Mapping with bowtie2---------------------------"
+#"index" is the prefix of the output filenames, not a directory
+echo "#### Building index: may take a while ####"
+#indexes are kept in the working directory so you can keep them if you want, as they take a while to generate
+cd ../
+bowtie2-build "$newgenome" index
+echo "#### Mapping results: ####"
+#bowtie2 arguments:
+#-k 1 means that bowtie2 will only report the best alignment for each variant (default behaviour)
+#-f specifies that the reads are in fasta format
+#-x [index] is the prefix of the index files
+#samtools view arguments:
+#-b: output in BAM format
+#-S: used to specifiy input is SAM format, but only in older versions of samtools
+bowtie2 -k 1 -f -x index "$TMPDIR"/variant_reads.out.fa | samtools view -bS - > "$TMPDIR"/reads_aligned.bam
```

Comment:
It would be helpful for readability to separate different subsections of this section with single empty lines, as you did in the previous section